### ISSUE TITLE
Can I speedup unit tests that produce and then consume

### ISSUE THREAD

**Issue by advanderveer** [twmb/franz-go#208]
First off, thank you for this high quality library! It has been a real joy to work with. 

I've been trying to setup a unit testing setup in which want each test to be isolated and operate on a dedicated topic and consume from it. Ofcourse it would be ideal if each test runs as quickly as possible but it seems that when I produce and consume right after each other there is always at least a 3s delay. It is always so close to the 3s that I suspect it is by design and I hope to configure it down somewhere. 

I've tried all duration configurations that this library offers but I can't really get the delay to go down. Maybe I've missed an option? Or, maybe it is a server configuration I can tweak? I'm using RedPanda, and it is just in a local docker container so can tweak anything I like.

For illustration, below is the testing code I'm trying to speed up. There is always exactly a 3s delay between "Fetching" and "Fetched". 

```Go
package stream_test

import (
	"context"
	"fmt"
	"testing"
	"time"

	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
	"go.uber.org/zap"

	"github.com/twmb/franz-go/pkg/kadm"
	"github.com/twmb/franz-go/pkg/kgo"
	"github.com/twmb/franz-go/plugin/kzap"
)

func TestStreamv2(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "stream")
}

var _ = Describe("consumer per test", func() {
	var ctx context.Context
	var cl *kgo.Client
	var adm *kadm.Client
	var topic string

	BeforeEach(func() {
		ctx, topic = context.Background(), fmt.Sprintf("tmp_changes_%d", time.Now().UnixNano())
		logs, err := zap.NewDevelopment()
		Expect(err).ToNot(HaveOccurred())
		cl, err = kgo.NewClient(
			kgo.SeedBrokers("localhost:9092"),
			kgo.ConsumerGroup("my-group"),
			kgo.ConsumeTopics(topic),
			kgo.WithLogger(kzap.New(logs)),
			kgo.DefaultProduceTopic(topic),
			kgo.AllowAutoTopicCreation(),
			kgo.MaxBufferedRecords(1),
                        kgo.FetchMinBytes(1),
		)
		Expect(err).ToNot(HaveOccurred())
		DeferCleanup(cl.Close)

		adm = kadm.NewClient(cl)
		DeferCleanup(func() {
			_, err = adm.DeleteTopics(ctx, topic)
			Expect(err).ToNot(HaveOccurred())
		})
	})

	It("should produce and consume", func() {
		for i := 0; i < 100; i++ {
			msg := &kgo.Record{}
			msg.Key = []byte(fmt.Sprintf("key-%d", i))
			msg.Value = []byte(fmt.Sprintf("value-%d", i))

			res := cl.ProduceSync(ctx, msg)
			Expect(res.FirstErr()).ToNot(HaveOccurred())
		}

		fmt.Println("Fetching...")
		fetches := cl.PollFetches(ctx)
		fmt.Println("Fetched...")
		if errs := fetches.Errors(); len(errs) > 0 {
			panic(fmt.Sprint(errs))
		}

		Expect(fetches.NumRecords()).To(Equal(100))
	})
})
```

The logging output might look something like below. The 3s wait is captured in the log line `2022-09-24T16:27:49.739+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read JoinGroup v5	{"broker": "0", "bytes_read": 229, "read_wait": "57.75µs", "time_to_read": "3.010023792s", "err": null}`
```
Running Suite: stream - /Users/adam/Documents/Projects/github.com/crewlinker/core/streamv2
==========================================================================================
Random Seed: 1664029665

Will run 1 of 1 specs
2022-09-24T16:27:46.493+0200	INFO	kzap@v1.1.1/kzap.go:110	beginning autocommit loop	{"group": "my-group"}
2022-09-24T16:27:46.493+0200	INFO	kzap@v1.1.1/kzap.go:110	immediate metadata update triggered	{"why": "client initialization"}
2022-09-24T16:27:46.493+0200	INFO	kzap@v1.1.1/kzap.go:110	producing to a new topic for the first time, fetching metadata to learn its partitions	{"topic": "tmp_changes_1664029666492478000"}
2022-09-24T16:27:46.493+0200	DEBUG	kzap@v1.1.1/kzap.go:106	opening connection to broker	{"addr": "localhost:9092", "broker": "seed 0"}
2022-09-24T16:27:46.495+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection opened to broker	{"addr": "localhost:9092", "broker": "seed 0"}
2022-09-24T16:27:46.495+0200	DEBUG	kzap@v1.1.1/kzap.go:106	issuing api versions request	{"broker": "seed 0", "version": 3}
2022-09-24T16:27:46.495+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote ApiVersions v3	{"broker": "seed 0", "bytes_written": 31, "write_wait": "30.667µs", "time_to_write": "12.333µs", "err": null}
2022-09-24T16:27:46.500+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read ApiVersions v3	{"broker": "seed 0", "bytes_read": 219, "read_wait": "41.542µs", "time_to_read": "5.233708ms", "err": null}
2022-09-24T16:27:46.500+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection initialized successfully	{"addr": "localhost:9092", "broker": "seed 0"}
2022-09-24T16:27:46.500+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Metadata v7	{"broker": "seed 0", "bytes_written": 55, "write_wait": "7.061958ms", "time_to_write": "11.292µs", "err": null}
2022-09-24T16:27:46.709+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Metadata v7	{"broker": "seed 0", "bytes_read": 164, "read_wait": "100.875µs", "time_to_read": "208.146625ms", "err": null}
2022-09-24T16:27:46.709+0200	INFO	kzap@v1.1.1/kzap.go:110	done waiting for metadata for new topic	{"topic": "tmp_changes_1664029666492478000"}
2022-09-24T16:27:46.709+0200	INFO	kzap@v1.1.1/kzap.go:110	beginning to manage the group lifecycle	{"group": "my-group"}
2022-09-24T16:27:46.709+0200	INFO	kzap@v1.1.1/kzap.go:110	joining group	{"group": "my-group"}
2022-09-24T16:27:46.709+0200	INFO	kzap@v1.1.1/kzap.go:110	initializing producer id
2022-09-24T16:27:46.709+0200	DEBUG	kzap@v1.1.1/kzap.go:106	opening connection to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.712+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection opened to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.712+0200	DEBUG	kzap@v1.1.1/kzap.go:106	issuing api versions request	{"broker": "0", "version": 3}
2022-09-24T16:27:46.712+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote ApiVersions v3	{"broker": "0", "bytes_written": 31, "write_wait": "43.5µs", "time_to_write": "60.459µs", "err": null}
2022-09-24T16:27:46.718+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read ApiVersions v3	{"broker": "0", "bytes_read": 219, "read_wait": "134.541µs", "time_to_read": "6.139834ms", "err": null}
2022-09-24T16:27:46.718+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection initialized successfully	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.719+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote InitProducerID v1	{"broker": "0", "bytes_written": 23, "write_wait": "9.274958ms", "time_to_write": "16.125µs", "err": null}
2022-09-24T16:27:46.719+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote FindCoordinator v2	{"broker": "0", "bytes_written": 28, "write_wait": "8.788208ms", "time_to_write": "20.583µs", "err": null}
2022-09-24T16:27:46.722+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read InitProducerID v1	{"broker": "0", "bytes_read": 24, "read_wait": "209.959µs", "time_to_read": "2.69675ms", "err": null}
2022-09-24T16:27:46.722+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read FindCoordinator v2	{"broker": "0", "bytes_read": 33, "read_wait": "2.937625ms", "time_to_read": "6.709µs", "err": null}
2022-09-24T16:27:46.722+0200	INFO	kzap@v1.1.1/kzap.go:110	producer id initialization success	{"id": 2013, "epoch": 0}
2022-09-24T16:27:46.722+0200	DEBUG	kzap@v1.1.1/kzap.go:106	opening connection to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.723+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection opened to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.723+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection initialized successfully	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.724+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote JoinGroup v5	{"broker": "0", "bytes_written": 132, "write_wait": "1.861708ms", "time_to_write": "16.042µs", "err": null}
2022-09-24T16:27:46.724+0200	DEBUG	kzap@v1.1.1/kzap.go:106	opening connection to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.725+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection opened to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.725+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection initialized successfully	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:46.725+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "3.068209ms", "time_to_write": "22.333µs", "err": null}
2022-09-24T16:27:46.729+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read JoinGroup v5	{"broker": "0", "bytes_read": 68, "read_wait": "174.916µs", "time_to_read": "5.0475ms", "err": null}
2022-09-24T16:27:46.729+0200	INFO	kzap@v1.1.1/kzap.go:110	join returned MemberIDRequired, rejoining with response's MemberID	{"group": "my-group", "member_id": "kgo-4d610cbe-39f8-4cdf-a18a-676cf773c958"}
2022-09-24T16:27:46.729+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote JoinGroup v5	{"broker": "0", "bytes_written": 172, "write_wait": "66.125µs", "time_to_write": "36.583µs", "err": null}
2022-09-24T16:27:46.732+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "43.25µs", "time_to_read": "7.005958ms", "err": null}
2022-09-24T16:27:46.732+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{0=>1}]"}
2022-09-24T16:27:46.732+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "100.833µs", "time_to_write": "17.875µs", "err": null}
2022-09-24T16:27:46.736+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "27.792µs", "time_to_read": "3.752208ms", "err": null}
2022-09-24T16:27:46.736+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{1=>2}]"}
2022-09-24T16:27:46.736+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "34.042µs", "time_to_write": "10.667µs", "err": null}
2022-09-24T16:27:46.739+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "46.333µs", "time_to_read": "2.97525ms", "err": null}
2022-09-24T16:27:46.739+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{2=>3}]"}
2022-09-24T16:27:46.739+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "18.417µs", "time_to_write": "11.75µs", "err": null}
2022-09-24T16:27:46.741+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "43.208µs", "time_to_read": "1.576083ms", "err": null}
2022-09-24T16:27:46.741+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{3=>4}]"}
2022-09-24T16:27:46.741+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "18.834µs", "time_to_write": "10.291µs", "err": null}
2022-09-24T16:27:46.743+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "39.792µs", "time_to_read": "1.795708ms", "err": null}
2022-09-24T16:27:46.743+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{4=>5}]"}
2022-09-24T16:27:46.743+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "47.5µs", "time_to_write": "10µs", "err": null}
2022-09-24T16:27:46.745+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "45µs", "time_to_read": "1.836666ms", "err": null}
2022-09-24T16:27:46.745+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{5=>6}]"}
2022-09-24T16:27:46.745+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "24.458µs", "time_to_write": "11.458µs", "err": null}
2022-09-24T16:27:46.747+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "34µs", "time_to_read": "1.674959ms", "err": null}
2022-09-24T16:27:46.747+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{6=>7}]"}
2022-09-24T16:27:46.747+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "16.083µs", "time_to_write": "22.833µs", "err": null}
2022-09-24T16:27:46.749+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "52.417µs", "time_to_read": "1.684ms", "err": null}
2022-09-24T16:27:46.749+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{7=>8}]"}
2022-09-24T16:27:46.749+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "18.166µs", "time_to_write": "17µs", "err": null}
2022-09-24T16:27:46.751+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "36.5µs", "time_to_read": "1.69525ms", "err": null}
2022-09-24T16:27:46.751+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{8=>9}]"}
2022-09-24T16:27:46.751+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Produce v7	{"broker": "0", "bytes_written": 154, "write_wait": "24.375µs", "time_to_write": "9.167µs", "err": null}
2022-09-24T16:27:46.753+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Produce v7	{"broker": "0", "bytes_read": 83, "read_wait": "42.541µs", "time_to_read": "1.970709ms", "err": null}
2022-09-24T16:27:46.753+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664029666492478000[0{9=>10}]"}
Fetching...

2022-09-24T16:27:49.739+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read JoinGroup v5	{"broker": "0", "bytes_read": 229, "read_wait": "57.75µs", "time_to_read": "3.010023792s", "err": null}
2022-09-24T16:27:49.739+0200	INFO	kzap@v1.1.1/kzap.go:110	joined, balancing group	{"group": "my-group", "member_id": "kgo-4d610cbe-39f8-4cdf-a18a-676cf773c958", "instance_id": null, "generation": 7, "balance_protocol": "cooperative-sticky", "leader": true}
2022-09-24T16:27:49.739+0200	INFO	kzap@v1.1.1/kzap.go:110	balancing group as leader
2022-09-24T16:27:49.739+0200	INFO	kzap@v1.1.1/kzap.go:110	balance group member	{"id": "kgo-4d610cbe-39f8-4cdf-a18a-676cf773c958", "interests": "interested topics: [tmp_changes_1664029666492478000], previously owned: "}
2022-09-24T16:27:49.740+0200	INFO	kzap@v1.1.1/kzap.go:110	balanced	{"plan": "kgo-4d610cbe-39f8-4cdf-a18a-676cf773c958{tmp_changes_1664029666492478000[0]}"}
2022-09-24T16:27:49.740+0200	INFO	kzap@v1.1.1/kzap.go:110	syncing	{"group": "my-group", "protocol_type": "consumer", "protocol": "cooperative-sticky"}
2022-09-24T16:27:49.740+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote SyncGroup v3	{"broker": "0", "bytes_written": 176, "write_wait": "56.333µs", "time_to_write": "21.583µs", "err": null}
2022-09-24T16:27:49.746+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read SyncGroup v3	{"broker": "0", "bytes_read": 69, "read_wait": "98.667µs", "time_to_read": "5.667ms", "err": null}
2022-09-24T16:27:49.746+0200	INFO	kzap@v1.1.1/kzap.go:110	synced	{"group": "my-group", "assigned": "tmp_changes_1664029666492478000[0]"}
2022-09-24T16:27:49.746+0200	INFO	kzap@v1.1.1/kzap.go:110	new group session begun	{"group": "my-group", "added": "tmp_changes_1664029666492478000[0]", "lost": ""}
2022-09-24T16:27:49.746+0200	INFO	kzap@v1.1.1/kzap.go:110	beginning heartbeat loop	{"group": "my-group"}
2022-09-24T16:27:49.746+0200	DEBUG	kzap@v1.1.1/kzap.go:106	entering OnPartitionsAssigned	{"with": {"tmp_changes_1664029666492478000":[0]}}
2022-09-24T16:27:49.746+0200	DEBUG	kzap@v1.1.1/kzap.go:106	sharded request	{"destinations": ["0"]}
2022-09-24T16:27:49.747+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote OffsetFetch v7	{"broker": "0", "bytes_written": 68, "write_wait": "87.125µs", "time_to_write": "24.417µs", "err": null}
2022-09-24T16:27:49.749+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read OffsetFetch v7	{"broker": "0", "bytes_read": 71, "read_wait": "73.667µs", "time_to_read": "2.641125ms", "err": null}
2022-09-24T16:27:49.750+0200	INFO	kzap@v1.1.1/kzap.go:110	assigning partitions	{"why": "newly fetched offsets for group my-group", "how": "assigning everything new, keeping current assignment", "input": "tmp_changes_1664029666492478000[0{-2 e-1 ce0}]"}
2022-09-24T16:27:49.750+0200	DEBUG	kzap@v1.1.1/kzap.go:106	assign requires loading offsets
2022-09-24T16:27:49.750+0200	INFO	kzap@v1.1.1/kzap.go:110	metadata update triggered	{"why": "loading offsets in new session from assign"}
2022-09-24T16:27:49.750+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Metadata v7	{"broker": "0", "bytes_written": 55, "write_wait": "34.667µs", "time_to_write": "16.5µs", "err": null}
2022-09-24T16:27:49.752+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Metadata v7	{"broker": "0", "bytes_read": 164, "read_wait": "69.625µs", "time_to_read": "2.43625ms", "err": null}
2022-09-24T16:27:49.753+0200	DEBUG	kzap@v1.1.1/kzap.go:106	metadata refresh has identical topic partition data	{"topic": "tmp_changes_1664029666492478000", "partition": 0, "leader": 0, "leader_epoch": 1}
2022-09-24T16:27:49.753+0200	DEBUG	kzap@v1.1.1/kzap.go:106	metadata refresh has identical topic partition data	{"topic": "tmp_changes_1664029666492478000", "partition": 0, "leader": 0, "leader_epoch": 1}
2022-09-24T16:27:49.753+0200	DEBUG	kzap@v1.1.1/kzap.go:106	offsets to load broker	{"broker": 0, "load": {"List":{"tmp_changes_1664029666492478000":{"0":{"At":-2,"Epoch":-1,"CurrentEpoch":1}}},"Epoch":null}}
2022-09-24T16:27:49.753+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote ListOffsets v4	{"broker": "0", "bytes_written": 79, "write_wait": "28.791µs", "time_to_write": "17.542µs", "err": null}
2022-09-24T16:27:49.756+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read ListOffsets v4	{"broker": "0", "bytes_read": 79, "read_wait": "69.958µs", "time_to_read": "2.517709ms", "err": null}
2022-09-24T16:27:49.756+0200	DEBUG	kzap@v1.1.1/kzap.go:106	opening connection to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:49.756+0200	DEBUG	kzap@v1.1.1/kzap.go:106	handled list results	{"broker": "0", "using": {"tmp_changes_1664029666492478000":{"0":{"Offset":0,"LeaderEpoch":1}}}, "reloading": {}}
2022-09-24T16:27:49.758+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection opened to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:49.758+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection initialized successfully	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:49.758+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Fetch v11	{"broker": "0", "bytes_written": 117, "write_wait": "2.299291ms", "time_to_write": "22.042µs", "err": null}
2022-09-24T16:27:49.763+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Fetch v11	{"broker": "0", "bytes_read": 901, "read_wait": "57.458µs", "time_to_read": "4.610709ms", "err": null}
Fetched...
2022-09-24T16:27:49.763+0200	DEBUG	kzap@v1.1.1/kzap.go:106	updated uncommitted	{"group": "my-group", "to": "tmp_changes_1664029666492478000[0{0=>0=>10}]"}
2022-09-24T16:27:49.763+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote Fetch v11	{"broker": "0", "bytes_written": 117, "write_wait": "27.958µs", "time_to_write": "37µs", "err": null}
2022-09-24T16:27:49.763+0200	DEBUG	kzap@v1.1.1/kzap.go:106	opening connection to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:49.765+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection opened to broker	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:49.765+0200	DEBUG	kzap@v1.1.1/kzap.go:106	connection initialized successfully	{"addr": "0.0.0.0:9092", "broker": "0"}
2022-09-24T16:27:49.765+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote DeleteTopics v3	{"broker": "0", "bytes_written": 58, "write_wait": "1.494125ms", "time_to_write": "17.333µs", "err": null}
2022-09-24T16:27:49.770+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read DeleteTopics v3	{"broker": "0", "bytes_read": 51, "read_wait": "66.042µs", "time_to_read": "5.427333ms", "err": null}
2022-09-24T16:27:49.770+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read Fetch v11	{"broker": "0", "bytes_read": 101, "read_wait": "58.125µs", "time_to_read": "7.109584ms", "err": null}
2022-09-24T16:27:49.770+0200	INFO	kzap@v1.1.1/kzap.go:110	assigning partitions	{"why": "invalidating all assignments in LeaveGroup", "how": "unassigning everything", "input": ""}
2022-09-24T16:27:49.770+0200	INFO	kzap@v1.1.1/kzap.go:110	heartbeat errored	{"group": "my-group", "err": "context canceled"}
2022-09-24T16:27:49.770+0200	DEBUG	kzap@v1.1.1/kzap.go:106	entering OnPartitionsRevoked	{"with": {"tmp_changes_1664029666492478000":[0]}}
2022-09-24T16:27:49.770+0200	DEBUG	kzap@v1.1.1/kzap.go:106	in CommitOffsetsSync	{"group": "my-group", "with": null}
2022-09-24T16:27:49.771+0200	DEBUG	kzap@v1.1.1/kzap.go:106	left CommitOffsetsSync	{"group": "my-group"}
2022-09-24T16:27:49.771+0200	INFO	kzap@v1.1.1/kzap.go:110	assigning partitions	{"why": "clearing assignment at end of group management session", "how": "unassigning everything", "input": ""}
2022-09-24T16:27:49.771+0200	INFO	kzap@v1.1.1/kzap.go:110	leaving group	{"group": "my-group", "member_id": "kgo-4d610cbe-39f8-4cdf-a18a-676cf773c958"}
2022-09-24T16:27:49.771+0200	DEBUG	kzap@v1.1.1/kzap.go:106	wrote LeaveGroup v3	{"broker": "0", "bytes_written": 75, "write_wait": "18.291µs", "time_to_write": "41.459µs", "err": null}
2022-09-24T16:27:49.773+0200	DEBUG	kzap@v1.1.1/kzap.go:106	read LeaveGroup v3	{"broker": "0", "bytes_read": 64, "read_wait": "53.625µs", "time_to_read": "2.171291ms", "err": null}
```

**Comment by godwhoa** [comment]
Not related to franz but worth mentioning, if the tests are truly isolated, you can run them in parallel to save time.

**Comment by advanderveer** [comment]
@godwhoa Thank you, that was planned for sure I just wanted to optimise the serial case as well. 

Also, in meantime I've figured out the setting that was delaying it. It was the broker setting described here: https://github.com/redpanda-data/redpanda/discussions/6530

**Comment by twmb** [comment]
Glad I waited until today to look at this 😛 

As you've correctly found, this is due to a configurable property within the brokers. Kafka has the same property (and the group code within Redpanda borrowed a lot of how group rebalancing works). There is nothing that can be done within a client to adjust this -- it must be done broker side.

Closing for now but please don't hesitate to ask any more questions

**Comment by advanderveer** [comment]
@twmb Ha, no problem. I actually ran into another suspicious timeout (2s this time). I've described in a comment over here: https://github.com/redpanda-data/redpanda/discussions/6530

This one might be a client issue because it emits the following on slow runs: 
```2022-09-26T20:21:05.899+0200	DEBUG	kzap@v1.1.1/kzap.go:106	produced	{"broker": "0", "to": "tmp_changes_1664216465786854000[0{retrying@-1,1(NOT_LEADER_FOR_PARTITION: This server is not the leader for that topic-partition.)}]"}``` 

Is the client waiting on the server appointed the "leader" of the partition (even though I'm just running it as a single node). Maybe I can configure the client to recheck this quicker? I've looked for a quick Opt but it seems I can't really make retries go faster, if that would even work?

Again, sorry if this is a broker setting. But it is hard for me to know.
